data_model/src/query.rs

```diff
@@ -1280,52 +1292,28 @@ pub mod http {
             /// The filter applied to the result on the server-side.
             pub filter: PredicateBox,
         }
-    }
 
-    model! {
-        /// I/O ready structure to send queries.
-        #[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
-        #[serde(transparent)]
-        #[repr(transparent)]
-        pub struct QueryRequest {
-            /// Payload
-            pub payload: Payload,
-        }
-    }
-
-    declare_versioned_with_scale!(VersionedSignedQueryRequest 1..2, Debug, Clone, iroha_macro::FromVariant, IntoSchema);
-
-    model! {
         /// I/O ready structure to send queries.
         #[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize, IntoSchema)]
-        #[version_with_scale(n = 1, versioned = "VersionedSignedQueryRequest")]
-        pub struct SignedQueryRequest {
+        #[version_with_scale(n = 1, versioned = "VersionedSignedQuery")]
+        pub struct SignedQuery {
             /// Payload
-            pub payload: Payload,
+            pub payload: QueryPayload,
             /// Signature of the client who sends this query.
-            pub signature: SignatureOf<Payload>,
+            pub signature: SignatureOf<QueryPayload>,
         }
-    }
-
-    declare_versioned_with_scale!(VersionedQueryResult 1..2, Debug, Clone, iroha_macro::FromVariant, IntoSchema);
 
-    model! {
         /// Sized container for all possible Query results.
         #[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
         #[version_with_scale(n = 1, versioned = "VersionedQueryResult")]
         #[serde(transparent)]
         #[repr(transparent)]
+        // TODO: This should be a separate type, not just wrap Value because it infects Value
+        // with variants that can only ever be returned, i.e. can't be used in instructions
+        // enum QueryResult { ... }
```

Comment:
Create a ticket for it. It could be problematic, because then the SDK would need to double all of their `Value`-based codegen. With that said, if there's no codegen, there's no problem.

Reply:
opened #3344